src/ol_infrastructure/infrastructure/aws/opensearch/__main__.py

```diff
@@ -39,27 +39,65 @@
 cluster_instance_type = search_config.get("instance_type") or "t3.medium.elasticsearch"
 disk_size = search_config.get_int("disk_size_gb") or 30  # noqa: WPS432
 
-search_security_group = aws.ec2.SecurityGroup(
-    "opensearch-security-group",
-    name_prefix=f"{environment_name}-opensearch-",
-    tags=aws_config.merged_tags({"Name": f"{environment_name}-opensearch"}),
-    description="Grant access to the OpenSearch service domain",
-    egress=default_egress_args,
-    ingress=[
+##########
+# CREATE #
+##########
+
+# Networking
+if search_config.get_bool("public_web"):
```

Comment:
I would initialize a variable of `is_public_web` or similar with the value of the config fetch so that it's not necessary to repeat that everywhere.